### PR TITLE
TypeScript extension and language server

### PR DESCRIPTION
**Pull request by svenefftinge**

### CODE REVIEW

File: packages/typescript/src/browser/typescript-client-contribution.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        @inject(Languages) protected readonly languages: Languages,
+        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory
+    ) {
+        super(workspace, languages, languageClientFactory);
+        registerTypeScript();
+    }
+
+    protected get globPatterns() {
+        return [
+            '**/*.ts',
```
Comment (akosyakov) [inline packages/typescript/src/browser/typescript-client-contribution.ts]: What about javascript? I think it was covered by the typescript package as well.

**Comment by akosyakov** [comment]
The last commit is not signed off.

File: packages/typescript/src/browser/typescript-client-contribution.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/typescript/src/browser/typescript-client-contribution.ts]: Should we add it to the typescript extension? It felt a bit odd. OTOH there are lots of similarities :).

Comment (akosyakov) [reply packages/typescript/src/browser/typescript-client-contribution.ts]: in VS code it seems to be 2 different extensions, I wonder how they handle the case when tsconfig allow compiling js files

File: packages/typescript/extension.package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@theia/typescript",
+  "version": "0.1.1",
+  "description": "Theia - Typescript Extension",
+  "dependencies": {
+    "@theia/core": "^0.1.1",
+    "@theia/languages": "^0.1.1",
+    "monaco-typescript": "^2.2.0",
```
Comment (akosyakov) [inline packages/typescript/extension.package.json]: it should depend on `@theia/monaco` to make sure that monaco is loaded before

**Comment by hexa00** [comment]
Just a note I think this should have been squashed...

**Pull request merged**